### ISSUE TITLE
Support alternative set of groups at breakouts creation

### ISSUE THREAD

**Issue by nidico** [bigbluebutton/bigbluebutton#14539]
Splitting large groups in breakout rooms takes quite some time which isn't always available for a presenter which is moderator at the same time. This is in particular bad if multiple breakout room sessions with different user assignements will be executed within one larger session.

Therefore we propose to allow moderators to assign (multiple) tags/labels to users and use those tags/labels to fill breakout room slots.

**Comment by ffdixon** [comment]
Are the users coming in from an LMS with the ability to assign groups?  

If so, another option would be to pass along the group assignment from the LMS when each user joins, and then use that assignment to populate users into breakout rooms.

**Comment by nidico** [comment]
> Are the users coming in from an LMS with the ability to assign groups?

In most cases not, in one case this might work (though there would be practical questions), so in general I'd say it should be done within BBB first.

**Comment by antobinary** [comment]
This is already implemented, see https://github.com/bigbluebutton/bigbluebutton/pull/13678#issuecomment-975567433

It is one of the BBB 2.5 Breakouts enhancements. BBB 2.5 is still in alpha stage.

I just realized we did not document this enhancement. A pull request to the docs is coming shortly. It will be the same information from the comment linked above.

**Comment by antobinary** [comment]
Merging into https://github.com/bigbluebutton/bigbluebutton/issues/13628

**Comment by nidico** [comment]
> Merging into #13628

I'm fine with merging both issues as they very much overlap, however I wonder whether what has been implemented in #13678 allows moderators to assign users to multiple groups (which can be used in subsequent breakout rooms) without prepopulating through an external LMS, as this is what this issue (#14539) is about in the first place. If this isn't fully implemented yet, we should keep the issue open.

**Comment by antobinary** [comment]
> assign users to multiple groups (which can be used in subsequent breakout rooms)

Based on putting `(multiple)` in parenthesis it looked like it was more of an optional aspect.

We now have documented the already implemented approach: https://github.com/bigbluebutton/bigbluebutton.github.io/pull/367

If I understand your question correctly, it can be boiled down to - can Johny be part of both TeamGreen and Assignment2Group so that the instructor can have the choice of creating a breakout based on TeamGreen **or** Assignment2Group as part of the create breakouts dialog...

I will let @gustavotrott answer whether we could support an extra level in the structure and possibly having Johny's id in both groups...

If I misunderstood, please clarify a bit more

**Comment by nidico** [comment]
Yeah, that's it. Within one session there are multiple subsequent breakout room sessions with different user assignments, so if `user1` is in `group1` and `group2`, then e.g. in the first session, `group1` could be used whereas in the second session `group2` is used.

**Comment by antobinary** [comment]
I will reopen this issue. But I have to say also that this sounds a bit like a niche issue. It would be really beneficial if other users would be able to chime in and say if this is indeed more common than it sounds.

Implementing support for this on BBB would also likely add a level of compexity at breakout creation time, something which we were trying to simplify and in BBB 2.5 to make it easier and faster to achieve.

In 2.5 we also included logic to try to group users together so that every time a breakout session is started, the groups are maintained from the previous setup (within that session -- i.e. that class)

That being said, personally I'd be hoping to see an actual LMS use the BBB 2.5 setup, following the specification in https://github.com/bigbluebutton/bigbluebutton/issues/13628#issuecomment-959673082 and only then iterate for future BBB based on a pattern handled by LMS.

Let's rename this issue to "Support alternative set of groups at breakouts creation' if that phrasing matches your request

**Comment by nidico** [comment]
> Let's rename this issue to "Support alternative set of groups at breakouts creation' if that phrasing matches your request

Done. That's fine and fits with the current state of development. (I wasn't aware of the enhancements in 2.5 when I created the issue.)

**Comment by nidico** [comment]
@stale bot: please reopen.

**Comment by nidico** [comment]
@antobinary : Could you please reopen? I missed the right time to stop the stale bot from performing destructive action and now [it's not possible anymore](https://github.com/actions/stale/issues/345).

(OT: Sorry for the strong words, but I really hate that bot or any other tool which automatically closes issues based on lack of activity - this behavior just drives contributors crazy!)

**Comment by gustavotrott** [comment]
What you are suggesting is something like predefined scenarios of Breakout rooms, right?
So when the Moderator open the modal to create breakout rooms, he would choose which scenario:
Scenario 1: Group X (User 1 and User 2, Group Y (User 3)
Scenario 2: Group DDD (User 1 and User 3), Group GGG (User 2)
Not sure if it is an usual user case...

I would rather prefer to suggest some way to use the current way to define the groups, and create some way to overwrite it during a live meeting... through an API endpoint or Smart Layout.. not sure the best way!

How do you think would be an easy way to achieve it without adding complexity to the interface, @nidico?
